V1.0 To Develop

manishmowall opened the pull request.

shubhampatil-webonise committed:
implement basic flow for application

shubhampatil-webonise committed:
add github oauth callback controller

shubhampatil-webonise committed:
update readme.md

shirish-babbur committed:
Add session-feature

shirish-babbur committed:
Add session-feature

Commit:
Add Http Request/Response Services

add service interface HttpRequestResponse and
its implementation OkhttpRequestResponse using Okhttp library

shubhampatil-webonise committed:
add complete authentication flow for login

Commit:
Add JsonParser feature

Commit:
Impelement JsonParser And Updates

remove Application class
remove JsonParserConfig class
remove PullRequestController class
update JsonParser and JsonParserImpl classes

shirish-babbur committed:
Add Session Service
provide session as a service by implementing wrapper over HttpSession

shirish-babbur committed:
Integrating Session Service with Oauth
- Add SessionService and SessionServiceImpl files

shubhampatil-webonise committed:
add session management in oauth cycle

shubhampatil-webonise committed:
update readme.md

Commit:
add new method to JsonParser-feature

Commit:
Make Change To JsonParserImpl class

made change to parse(String,String) function

Commit:
Add addToJson Method To JsonParser Class

Commit:
Modified JsonParserService

shubhampatil-webonise committed:
Implement fetching repo list from sql database using Spring Data JPA
- add a database service which can be used for both sql and mongo dbs

shubhampatil-webonise committed:
add mongodb document models to project and test sample mongodb fetching using spring-data mongdb

shubhampatil-webonise committed:
rebased dashboard-feature with v1.0 and resolved conflicts

shubhampatil-webonise committed:
Merge pull request #3 from shubhampatil-webonise/dashboard-feature

Dashboard feature

shubhampatil-webonise committed:
Merge pull request #4 from shubhampatil-webonise/JsonParser-feature

Json parser feature

shubhampatil-webonise committed:
implement logic for displaying repository list on dashboard

- fetched repo list from sql using spring data jpa
- saved real time data from webhooks into mongo db using spring data mongodb
- sync between mongodb and mysql

shubhampatil-webonise committed:
refactor code

deeps-webonise committed:
Implemented mail service feature using SMTP

deeps-webonise committed:
Implemented mail service feature using SMTP

shubhampatil-webonise committed:
Merge pull request #5 from shubhampatil-webonise/repositoryList-feature

Repository list feature

deeps-webonise committed:
Refactor pom.xml

deeps-webonise committed:
Merge pull request #6 from shubhampatil-webonise/mailservice-feature

Mailservice feature

Commit:
Implementation Of Pull Request Webhook Services

- create PullRequestController class for webhooks related to PR.
- create PR services to model data from webhooks to be stored in DB.

There are four services and their implementations for each webhooks related to PR.

Commit:
Merge pull request #7 from shubhampatil-webonise/PRService-feature

PR service feature

shubhampatil-webonise committed:
add integration of PR webhooks with database

- implement logic for saving new PRs, Reviews , Review Comments and PR comments.

shubhampatil-webonise committed:
Merge pull request #8 from shubhampatil-webonise/PR-Database-Integration-Feature

add integration of PR webhooks with database

shirish-babbur committed:
Add implementation for integration of branch webhooks with database

Commit:
Merge pull request #9 from shubhampatil-webonise/branch-database-integration-feature

Add implementation for integration of branch webhooks with database

shirish-babbur committed:
Add implementation for integration of collaborator webhooks with database

Commit:
Merge pull request #10 from shubhampatil-webonise/collaborator-database-integration-feature

Add implementation for integration of collaborator webhooks with data…

shirish-babbur committed:
Add stale branch notification Service
- Add implementation of Stale Branch Detector logic
- Integrates Email service with Stale branch detector
- Add scheduler for sending mails to stale branch creators

deeps-webonise committed:
Merge pull request #11 from shubhampatil-webonise/stale-branch-detector-feature

Stale branch detector

shubhampatil-webonise committed:
add basic GUI to repository panel

shubhampatil-webonise committed:
add basic GUI to repository panel

shubhampatil-webonise committed:
resolve merge conflict

shubhampatil-webonise committed:
Update StaleBranchNotificationMail Class, css and html files

- changed cron time
- updated repository.css and repository.html file to show repo,pr, branch and collaborator details

Commit:
Merge pull request #12 from shubhampatil-webonise/repositoryGUI-feature

Repository gui feature

shubhampatil-webonise committed:
Refactor Code

- made all packages lowercase
- remove Services and ServiceImplementations package
- add service interfaces and its implementations in services package

ameyawebonise commented on src/main/java/org/webonise/gitmetrics/controllers/DashboardController.java:
```diff
@@ -0,0 +1,51 @@
+
+@Controller
+public class DashboardController {
+    private static final Logger logger = Logger.getLogger(DashboardController.class.getName());
```
why the `getName()` why not `DashboardController.class`?

ameyawebonise replied:
I see this pattern everywhere.why so?

ameyawebonise commented on src/main/java/org/webonise/gitmetrics/controllers/webhooks/BranchController.java:
```diff
@@ -0,0 +1,30 @@
+@RequestMapping("/webhooks/repository/branch")
+public class BranchController {
+    @Autowired
+    private BranchService service;
```
name the variable to `branchService`.Makes it easy when a controller has to call many services

ameyawebonise commented on src/main/java/org/webonise/gitmetrics/documents/Owner.java:
```diff
@@ -0,0 +1,14 @@
+
+@Document
+public class Owner {
+    public String login;
```
why do you have the properties to be `public`? and then have getters ?

ameyawebonise commented on src/main/java/org/webonise/gitmetrics/documents/Assignee.java:
```diff
@@ -0,0 +1,14 @@
+
+@Document
+public class Assignee {
+    public String login;
```
you are breaking encapsulation when you make the fields to be public.Please make them private!

ameyawebonise commented on src/main/java/org/webonise/gitmetrics/services/implementations/AuthorizationServiceImpl.java:
```diff
@@ -0,0 +1,50 @@
+            String organizationDetails = httpRequestResponseService.get("https://api.github.com/orgs/" + organization + "/memberships/" + loginHandle, headers);
+            String role = gson.fromJson(organizationDetails, JsonObject.class).get("role").getAsString();
+
+            if (role.equalsIgnoreCase("admin"))
```
why not `return role.equalsIgnoreCase("admin")`

ameyawebonise commented on src/main/java/org/webonise/gitmetrics/services/implementations/AuthorizationServiceImpl.java:
```diff
@@ -0,0 +1,50 @@
+                return true;
+
+        } catch (IOException e) {
+            logger.error(e.getMessage());
```
just dont log the e.message.Word the message properly and log the entire `e`.this preserves the stack trace so that debugging becomes easier

ameyawebonise commented on src/main/java/org/webonise/gitmetrics/services/implementations/DatabaseServiceImpl.java:
```diff
@@ -0,0 +1,493 @@
+                    try {
+                        pullRequest.set(entry.getKey(), entry.getValue());
+                    } catch (Exception e) {
+                        logger.error(e.getMessage());
```
better exception logging please

ameyawebonise commented on src/main/java/org/webonise/gitmetrics/services/implementations/DatabaseServiceImpl.java:
```diff
@@ -0,0 +1,493 @@
+    }
+
+    @Override
+    public void removeCommentFromPullRequest(String repositoryName, int pullRequestNumber, long commentId, Map<String, Object> updateKeys) {
```
why do you want to remove comment from pullrequest?

ameyawebonise commented on src/main/java/org/webonise/gitmetrics/services/implementations/DatabaseServiceImpl.java:
```diff
@@ -0,0 +1,493 @@
+                        currentReview.setBody(review.getBody());
+                        currentReview.setUser(review.getUser());
+                        currentReview.setSubmittedAt(review.getSubmittedAt());
+//                            currentReview.setComments(review.getComments());
```
dont check in commented code

ameyawebonise commented:
Guys,please do the following changes:
1.Your models break encapsulation.You need the props to be private and expose their values only through getters and setters.
2.Your models are missing the `equals()` and `hashCode()` methods.
3.Exceptions should be logged in the whole ,not only `e.getMessage()`

shubhampatil-webonise committed:
Made Following Changes

- Made all data member of models private and created respective getters and setters.
- created equals() and hashCode() in all models.
- logging stacktrace rather than message only.

deeps-webonise committed:
Updated README.md

Added Definition,information regarding webhooks,databases,environment variable and how to run it.

deeps-webonise committed:
Merge branch 'v1.0' of https://github.com/shubhampatil-webonise/GitMetrics into v1.0

deeps-webonise committed:
Update Stale Branch Notification code And modified PullRequest Model

- replace getField method to getDeclaredField method
- remove StaleBranch class  and StaleBranchDetector class
- rename StaleBranchNotificationMailer class to StaleBranchNotifier class

update code in StaleBranchNotifier to get PRs and Branches from database and take
commit detials from branch api call.

deeps-webonise committed:
Update Stale Branch Notification code And modified PullRequest Model

- replace getField method to getDeclaredField method
- remove StaleBranch class  and StaleBranchDetector class
- rename StaleBranchNotificationMailer class to StaleBranchNotifier class

update code in StaleBranchNotifier to get PRs and Branches from database and take
commit detials from branch api call.

Commit:
Create Member Service And update PullRequest Model And PullRequestReview Webhook

- create MemberService interface and its implementation MemberServiceImpl class.
- replace getField method to getDeclaredField method in set method of PullRequest model class.
- removed print method in actionOnSubmit method of PullRequestReviewImpl class.

shubhampatil-webonise committed:
add complete GUI for Repository details including PRs, Branches and Collaborators

shubhampatil-webonise committed:
Merge pull request #16 from shubhampatil-webonise/repositoryUI-feature

add complete GUI for Repository details including PRs, Branches and C…

shubhampatil-webonise committed:
Merge pull request #15 from shubhampatil-webonise/organizationmemberservice-feature

Create Member Service And update PullRequest Model And PullRequestReview Webhook

Commit:
Merge branch 'stalebranch-feauture' of github.com:shubhampatil-webonise/GitMetrics into stalebranch-feauture

Commit:
Merge pull request #14 from shubhampatil-webonise/stalebranch-feauture

Update Stale Branch Notification code And modified PullRequest Model

shubhampatil-webonise committed:
implement and modify GUI for complete application with rendering information of organization members

shubhampatil-webonise committed:
Merge pull request #17 from shubhampatil-webonise/membersGUI-feature

implement and modify GUI for complete application with rendering info…
